cmd/deploy/proxy.go

```diff
@@ -0,0 +1,182 @@
+package deploy
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/okteto/okteto/pkg/log"
+	"github.com/okteto/okteto/pkg/model"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+)
+
+type proxyConfig struct {
+	port         int
+	certificates []tls.Certificate
+	token        string
+}
+
+type proxy struct {
+	s           *http.Server
+	proxyConfig proxyConfig
+}
+
+func newProxy(proxyConfig proxyConfig) *proxy {
+	return &proxy{
+		proxyConfig: proxyConfig,
+	}
+}
+
+// Start starts the proxy server
+func (p *proxy) Start(ctx context.Context, name string, clusterConfig *rest.Config) error {
+	handler, err := p.createProxyHandler(ctx, name, clusterConfig)
+	if err != nil {
+		return err
+	}
+
+	p.createAndStartProxyServer(ctx, handler, p.proxyConfig.port, p.proxyConfig.certificates)
+
+	return nil
+}
+
+func (p *proxy) createAndStartProxyServer(ctx context.Context, handler http.Handler, port int, certs []tls.Certificate) {
+	s := &http.Server{
+		Addr:         fmt.Sprintf(":%d", port),
+		Handler:      handler,
+		ReadTimeout:  3600 * time.Second,
```

Comment:
I think these are too long and will probably cascading issues if something bad happens. I don't think the proxy will be open to much abuse but I figure the timeouts from https://blog.cloudflare.com/exposing-go-on-the-internet/ would be a good start.